compiler/passes/errorHandling.cpp

```diff
@@ -381,6 +390,27 @@ bool ErrorHandlingVisitor::enterCallExpr(CallExpr* node) {
     throwBlock->insertAtTail(new DefExpr(fixedError));
     throwBlock->insertAtTail(new CallExpr(PRIM_MOVE, fixedError, fixError));
 
+    if (!catchesStack.empty()) {
+      Expr*      parent      = throwBlock;
+      CatchStmt* parentCatch = NULL;
+      while (parentCatch == NULL) {
+        parent      = parent->parentExpr;
+        parentCatch = toCatchStmt(parent);
+      }
+
+      if (DefExpr* catchFilter = parentCatch->expr()) {
```

Comment:
This can be tricky, because we will cast the error variable (and create temporaries in the process). Setting one of these to nil won't affect the other copies of it. Can you make sure to add some tests that exercise this code? Are those tests already there? Or are they not yet in this PR for some reason?

Reply:
I've added `nil-check-catchall` and `nil-check-filter`, which exercise the code inside the conditional for named catchalls and filtered catches respectively. `throw-from-catch`, `throw-inner-catch`, and `unmarked-call-catch` ensure that the conditional is not entered for unnamed catchalls.

Reply:
Cool. I think this PR is ready now, thanks.